src/app/pages/SearchResultsPage/SearchResultsList.tsx

```diff
@@ -94,6 +98,35 @@ export const SearchResultsList: FC<{
           linkLabel={t('search.results.accounts.viewLink')}
         />
 
+        <ResultsGroupByType
+          title={t('search.results.accounts.title')}
+          results={searchResults.filter(
+            (item): item is AccountAddressResult => item.resultType === 'accountAddress',
+          )}
+          resultComponent={item =>
+            item.layer === Layer.consensus ? (
+              <DeferredConsensusAccountDetails
+                network={item.network}
+                address={item.address}
+                tokenPrices={tokenPrices}
+                highlightedPartOfName={searchTerm}
+                showLayer={true}
+              />
+            ) : (
+              <DeferredRuntimeAccountDetails
+                network={item.network}
+                layer={item.layer}
+                address={item.address}
+                tokenPrices={tokenPrices}
+                highlightedPartOfName={searchTerm}
+                showLayer={true}
+              />
+            )
+          }
+          link={acc => RouteUtils.getAccountRoute(acc, acc.address)}
+          linkLabel={t('search.results.accounts.viewLink')}
+        />
+
```

Comment:
Why do we need to reimplement the accounts results group?

Reply:
We don't necessarily have to reimplement the group, but we need to extend it at least, because this kind of search result can't (immediately) be displayed like other accounts, since we only have the address at this point. (That is what we have the `DeferrentRuntimeAccountDetails` component and it's consensus counterpart.) So, possible approaches here:

- Extend the existing accounts group with a renderer function that can handle both type of results (full accounts and addresses to be loaded later)
- Modify the search function itself so it also loads the accounts and returns them. This means we don't need to touch the displaying part. Problems with this approach:
   - It would make the whole search slower (no results would be displayed for a longer time)
   - It's not easy to conditionally call multiple hooks from within a hook, thanks to the rules of hooks. The best way to do that is to actually render multiple components, and have each component call it's own hook to load that data, just as we are doing now with the `DeferredRuntimeAccountDetails`   component (and friends)
   
I would probably go with the first approach.

Reply:
Second approach - getting full account objects makes a lot more sense to me:
- it's how an API endpoint would/will behave if we would have it. I expect API to search validators by name will return validator objects too
- I think it can be implemented in an easier to maintain way
- no need to call multiple hooks, just one conditional

Reply:
> no need to call multiple hooks, just one conditional

OK, so let's assume that the client-side search (in the address-name mappings) have determined the address of the 3 accounts that have a match for the given name fragment. So we have the 3 addresses. How do we lead the 3 accounts with a single nexus API call?

Reply:
It would make multiple API requests, but in a single useQuery hook

Reply:
Let's see if we can do something like [this](https://tanstack.com/query/latest/docs/framework/react/guides/dependent-queries?from=reactQueryV3#usequeries-dependent-query) in the nexus api  wrapper layer...

Reply:
oh I haven't seen that before, looks good too

Reply:
> It would make multiple API requests, but in a single useQuery hook

How would you do that?

The problem seems to be that we already have multiple layers of wrappers around the actual Nexus API call, and we have added come logic in each layers. Even the auto-generated layer adds useful stuff, and then there is the data transformation logic in the wrapper layer, plus we want to mix in the results of the RPC calls. Extracting all that logic, and re-applying it after we call the Nexus API another way (ie. avoiding our current wrappers) would be a pain. So I would stick to using the already existing hooks... but then already have `useQuery` hooks, and it's not easy to dynamically call them multiple times from within a single hook.

> it's how an API endpoint would/will behave if we would have it. I expect API to search validators by name will return validator objects too

Yeah, if we had that API, it would be a different matter ... but we don't.

For now, it seems to be a lot easier to just render more components, and have them load their own data using the existing APIs.

    * * *
 
As a first step I'll just unite the two "Accounts" section in the results list. (Although I don't think we can have both of them at the same time, given that you would have to search for completely different things to get results in the two sections.)

Reply:
> As a first step I'll just unite the two "Accounts" section in the results list. 

I have done that now.

Reply:
How about we limit the number of results from this specific sub-search? (Accounts listed because of matching name segment.) If we can agree to show at most 20 such result (or any fixed number), then we can launch the subsequent account loader sub-queries using a simple `for` cycle of fixed length, which doesn't break the rules of hooks. (And then  use `enabled` to selectively enable the ones we need, based on the number of matching accounts found.)

That way we can emulate that we have the missing API, and then all the higher layers (UI etc) can be kept simple. (We can drop the DefferedRuntimeAccount component and friends, no need for any expansion in the result list account section,etc.)

Maybe we can indicate at the end of the search results list if there are more matches that haven't been displayed. @donouwens would that be acceptable?

Reply:
>  where is the cut off

Wherever we want it to be. It can be anything, but it has to be a fixed number. Bigger numbers come with a bigger performance penalty, when we try to load all the account data for all matching accounts _all at once_, it as many parallel requests as we have accounts to display.

> would there be a way for users to view to additional results?

Not an easy one.

   * * *

Please note that we also have other (currently unclear an undefined) limits for all other results, too. I don't know what those are, but probably no more than 100 or 1000 for each kind of objects.

Reply:
> Second approach - getting full account objects makes a lot more sense to me

I have changed implementation so that we are now getting the full account objects with the hooks now. 
Accordingly, eliminated the new result type.

Please note that this should only be reviewed after #1422.

Reply:
As we have discussed at today's meeting, this is all we can do here, for now.